### PR TITLE
Misc fixes

### LINKED ISSUE #111: Escaped Unicode string not parsed correctly
Consider
```python
u = f'\N{GREEK CAPITAL LETTER DELTA}'
```
which is a ascii-only way to get the unicode-glyph with the name in between `{` and `}`.
Parse Tree:
```
expression_statement [0, 0] - [0, 37]
  assignment [0, 0] - [0, 37]
    left: identifier [0, 0] - [0, 1]
    right: string [0, 4] - [0, 37]
      interpolation [0, 8] - [0, 36]
        ERROR [0, 9] - [0, 29]
          identifier [0, 9] - [0, 14]
          identifier [0, 15] - [0, 22]
          identifier [0, 23] - [0, 29]
        identifier [0, 30] - [0, 35]
```

### LINKED ISSUE #119: `await` not parse properly after `in`
In this specific example, the `await` keyword isn't being properly highlighted:

```python
async def region_exists(region: str) -> bool:
    return region in await all_regions()
#                    ^^^^^ This keyword is not highlighted properly
```

Huh, it seems that GitHub also doesn't highlight it properly.

### LINKED ISSUE #123: Error on default tuple arguments
```
def comp_args((a, b)=(3, 4)):
    return a, b
```

```
(module [0, 0] - [2, 0]
  (function_definition [0, 0] - [1, 15]
    name: (identifier [0, 4] - [0, 13])
    parameters: (parameters [0, 13] - [0, 28]
      (ERROR [0, 14] - [0, 21]
        (tuple_pattern [0, 14] - [0, 20]
          (identifier [0, 15] - [0, 16])
          (identifier [0, 18] - [0, 19])))
      (tuple_pattern [0, 21] - [0, 27]
        (ERROR [0, 22] - [0, 26]
          (integer [0, 22] - [0, 23])
          (integer [0, 25] - [0, 26]))))
    body: (block [1, 4] - [1, 15]
      (return_statement [1, 4] - [1, 15]
        (expression_list [1, 11] - [1, 15]
          (identifier [1, 11] - [1, 12])
          (identifier [1, 14] - [1, 15]))))))
fail/DefaultTupleArguments.py	0 ms	(ERROR [0, 14] - [0, 21])
```

### LINKED ISSUE #124: Error on exec on variable
```
func = "print"
exec func in {}
```

```
(module [0, 0] - [1, 15]
  (expression_statement [0, 0] - [0, 14]
    (assignment [0, 0] - [0, 14]
      left: (identifier [0, 0] - [0, 4])
      right: (string [0, 7] - [0, 14])))
  (expression_statement [1, 0] - [1, 15]
    (comparison_operator [1, 0] - [1, 15]
      (identifier [1, 0] - [1, 4])
      (ERROR [1, 5] - [1, 9]
        (identifier [1, 5] - [1, 9]))
      (dictionary [1, 13] - [1, 15]))))
test.py	0 ms	(ERROR [1, 5] - [1, 9])
```

### LINKED ISSUE #125: Error on print not True
```
print not True
```

```
(module [0, 0] - [0, 14]
  (ERROR [0, 0] - [0, 9]
    (identifier [0, 0] - [0, 5]))
  (expression_statement [0, 10] - [0, 14]
    (true [0, 10] - [0, 14])))
test.py	0 ms	(ERROR [0, 0] - [0, 9])
```

### LINKED ISSUE #126: Error on backslash in function with no newline after it
From [here](https://github.com/JetBrains/intellij-community/blob/master/python/testData/formatter/trailingBlankLinesWithBackslashesAtFunctionEndNoNewLine.py).

```
def foo():
    pass \
\
\
```

```
(module [0, 0] - [3, 1]
  (ERROR [0, 0] - [1, 8]
    (identifier [0, 4] - [0, 7])
    (parameters [0, 7] - [0, 9])
    (pass_statement [1, 4] - [1, 8])))
fail/trailingBlankLinesWithBackslashesAtFunctionEndNoNewLine.py	0 ms	(ERROR [0, 0] - [1, 8])
```

### LINKED ISSUE #132: list_splat inside of expression_list cannot be parsed
The following produces parsing errors
```python
a,c = [1,2],3
w, x, y, z = 0, *a, c 
```

My proposed grammar change to resolve the issue:
```diff
diff --git a/grammar.js b/grammar.js
index bd0ee81..639b0f9 100644
--- a/grammar.js
+++ b/grammar.js
@@ -440,13 +440,13 @@ module.exports = grammar({
     ),
 
     expression_list: $ => prec.right(seq(
-      $.expression,
+      choice($.expression, $.list_splat),
       choice(
         ',',
         seq(
           repeat1(seq(
             ',',
-            $.expression
+            choice($.expression, $.list_splat)
           )),
           optional(',')
         ),
```

### LINKED ISSUE #202: Parsing f-strings with unpackings
Hi,

there is an issue in parsing f-strings with unpackings, see example:

```
f"{*a,}"
````
Results in
```
(module (expression_statement (string (interpolation (ERROR) (expression_list (identifier))))))
```

### LINKED ISSUE #207: Deadlock/memory leak/crash when parsing a specific file
Hello,
While I've parsed countless files so far without issue, when parsing [this very file](https://raw.githubusercontent.com/Tomi23R/sureBet-bot/294dda827eab8440fe06b97f698beabf6b580946/crawler/tenis_codere.py) tree-sitter deadlocks and starts consuming more and more memory (100s of GB) until there is no more left, and it crashes with core dumped.
Note that if I replace all the tabs `\t` by 4 spaces, then the parser however works just fine as usual.

**versions:**
- `tree-sitter==0.20.1`
- lib compiled from `tree-sitter-python` at tag `c6cfa75 4f25ce7` (master as of Feb 28th)

**repro:**
```python
import requests
from pathlib import Path
import tree_sitter as ts

# download code to parse
url = "https://raw.githubusercontent.com/Tomi23R/sureBet-bot/294dda827eab8440fe06b97f698beabf6b580946/crawler/tenis_codere.py"
res = requests.get(url)
code = res.content.decode("utf8")

# instantiate tree-sitter
lib_path = Path().resolve().absolute() / "tree-sitter" / "python.so"
assert lib_path.exists(), "Update the lib_path to your install"
language = ts.Language(lib_path, "python")
parser = ts.Parser()
parser.set_language(language)

# this does not deadlock:
parser.parse(code.replace("\t", "    ").encode("utf8"))

# this deadlocks:
parser.parse(code.encode("utf8"))
```

### LINKED ISSUE #210: Expose real string quotes in a tree to don't mislead
In a current grammar definition:
```js
    string: $ => seq(
      field('prefix', alias($._string_start, '"')),
      repeat(choice(
        field('interpolation', $.interpolation),
        field('string_content', $.string_content)
      )),
      field('suffix', alias($._string_end, '"'))
    ),
```
Aliases hide actual tokens like `'''` and `"""`:
```js
alias($._string_start, '"')
alias($._string_end, '"')
```

### LINKED ISSUE #212: Dedent regression in valid ternary expression in f-string
The valid expression
```python
def function():
     return f"""
{"string1" if True else
 "string2"}"""
 ```
 parses as 
 ```
 (module [0, 0] - [5, 0]
  (function_definition [0, 0] - [2, 23]
    name: (identifier [0, 4] - [0, 12])
    parameters: (parameters [0, 12] - [0, 14])
    (ERROR [1, 5] - [2, 23]
      string_content: (string_content [1, 16] - [2, 0])
      (string [2, 1] - [2, 10]
        string_content: (string_content [2, 2] - [2, 9]))
      (true [2, 14] - [2, 18]))
    body: (block [2, 23] - [2, 23]))
  (ERROR [3, 0] - [3, 13]
    (string [3, 0] - [3, 9]
      string_content: (string_content [3, 1] - [3, 8]))
    (ERROR [3, 9] - [3, 10])))
```
however, the similar expressions
```python
def function():
     return f"""
{"string1" if True else "string2"}"""
```
and
```python
f"""
{"string1" if True else
"string2"}"""
```
both parse correctly, implying the the problem is with the indentation.

`git bisect` reveals that the problem is with commit 188b6b0.
its parent 597c4bb parses correctly as 
```
(module [0, 0] - [4, 0]
  (function_definition [0, 0] - [3, 14]
    name: (identifier [0, 4] - [0, 12])
    parameters: (parameters [0, 12] - [0, 14])
    body: (block [1, 5] - [3, 14]
      (return_statement [1, 5] - [3, 14]
        (string [1, 12] - [3, 14]
          (interpolation [2, 0] - [3, 11]
            (conditional_expression [2, 1] - [3, 10]
              (string [2, 1] - [2, 10])
              (true [2, 14] - [2, 18])
              (string [3, 1] - [3, 10]))))))))
```
but commit 188b6b0 parses it as
```
(module [0, 0] - [4, 0]
  (function_definition [0, 0] - [2, 23]
    name: (identifier [0, 4] - [0, 12])
    parameters: (parameters [0, 12] - [0, 14])
    (ERROR [1, 5] - [2, 23]
      (string [2, 1] - [2, 10])
      (true [2, 14] - [2, 18]))
    body: (block [2, 23] - [2, 23]))
  (ERROR [3, 1] - [3, 14]
    (string [3, 1] - [3, 10])
    (ERROR [3, 10] - [3, 11])))
```

### PR DESCRIPTION
**Pull request by amaanq**
Closes #111 
Closes #119
Closes #123  
Closes #124
Closes #125 
Closes #126 
Closes #132
Closes #202 
Closes #210
Closes #212 

cc @ahlinc @stephen-huan

### CODE REVIEW

File: test/corpus/statements.txt [hunk 1]
```diff
@@ -671,102 +683,105 @@ async def d(a:str="default", b=c) -> None:
...
-    name: (identifier)
-    parameters: (parameters
+    (identifier)
+    (parameters
       (identifier)
       (default_parameter
-        name: (identifier)
-        value: (identifier))
+        (identifier)
+        (identifier))
```
Comment (maxbrunsfeld) [inline test/corpus/statements.txt:748]: It looks like the field names got removed from some of these expectations. Could you restore those?

File: test/corpus/expressions.txt [hunk 2]
```diff
@@ -833,40 +845,43 @@ lambda (a, b): (a, b)
...
-      (tuple
-        (identifier)
-        (identifier)))))
+	(lambda
+	  (lambda_parameters
+		(identifier)
+		(identifier))
+	  (call
+		(identifier)
+		(argument_list
```
Comment (maxbrunsfeld) [inline test/corpus/expressions.txt:854]: The indentation in this assertion is inconsistent with most of the others. Could you adjust this to use 2-space indentation?

**Review by maxbrunsfeld — COMMENTED** [review]
Looks good to me - just left a couple of comments.

File: test/corpus/statements.txt (hunk 1 above)
Comment (amaanq) [reply test/corpus/statements.txt]: done

File: test/corpus/expressions.txt (hunk 2 above)
Comment (amaanq) [reply test/corpus/expressions.txt]: done

**Comment by amaanq** [comment]
After this - a review of https://github.com/tree-sitter/tree-sitter-bash/pull/179 would be greatly appreciated too, thank you!!

**Comment by amaanq** [comment]
@aryx since this has Max's seal of approval (bar the changes that I fixed), do you mind merging this? :sweat_smile:

**Review by aryx — APPROVED** [review]
Would have been nice to add examples in the corpus for each issue you closed
(also would be nicer to make separate PRs for each fix).

**Comment by amaanq** [comment]
> Would have been nice to add examples in the corpus for each issue you closed
> (also would be nicer to make separate PRs for each fix).
> 

Yes, I plan to make my prs smaller from now on (targeting 2-3 issues max)

File: grammar.js [hunk 3]
```diff
@@ -55,9 +57,9 @@ module.exports = grammar({
     $._newline,
     $._indent,
     $._dedent,
-    $._string_start,
+    $.string_start,
     $._string_content,
```
Comment (wadeeedex-hash) [inline grammar.js:61]:
```suggestion
    $._string_content,
``` messages 


```suggestion
@dcreager     $._string_content,
```

File: grammar.js [hunk 4]
```diff
@@ -177,11 +179,11 @@ module.exports = grammar({
...
-      prec(-10, seq(
+      prec(-3, prec.dynamic(-1, seq(
         'print',
         commaSep1(field('argument', $.expression)),
         optional(',')
-      ))
+      ))),
     ),
 
     chevron: $ => seq(
```
Comment (wadeeedex-hash) [inline grammar.js:189]: for i in $(seq 1 3 10); do
  echo "Iteration: $i"
done

File: grammar.js [hunk 5]
```diff
@@ -55,9 +57,9 @@ module.exports = grammar({
     $._newline,
     $._indent,
     $._dedent,
-    $._string_start,
+    $.string_start,
     $._string_content,
-    $._string_end,
+    $.string_end,
```
Comment (wadeeedex-hash) [inline grammar.js:62]: 1.

File: test/corpus/statements.txt [hunk 6]
```diff
@@ -161,8 +161,14 @@ print >> a, "b", "c"
   (print_statement
     (chevron
```
Comment (wadeeedex-hash) [inline test/corpus/statements.txt:162]: ### 
```suggestion
    

- > [(chevron](url)

```

**Review by wadeeedex-hash — COMMENTED** [review]
Kk

**Pull request merged**